### DISCUSSION TITLE
Are there any good ways to increase performance?

### DISCUSSION THREAD

**Discussion by wogus3602** [pointfreeco/swift-composable-architecture#896] [Ideas]
I'm working on a project using TCA. There were no performance issues at first, but as the project size grew, a few issues became slow to respond.

Because I ran everything on the main thread, I thought the performance was bad, so I improved it by using subscribe(on:). Second, I used WithViewStore(store.stateless) where I did not use WithViewStore(store) and did not use state.

Do you have your own know-how to improve performance?

**Comment by eimantas** [comment]
If you have long navigation stacks, there is a chance that changes to the last view in stack causes all of the previous views to redraw (this is where the performance might degrade). One developer on the project I am working on noticed this and used PFCs provided "trick". You can read more [here](https://github.com/pointfreeco/swift-composable-architecture/blob/599a2398adaaa7a4e3f5420cde7728c39e33677e/Sources/ComposableArchitecture/Store.swift#L243).

In a gist — isolating a state from the app domain into its own should help with needless view refreshes.
